src/MSBuild/TerminalLogger/TerminalLogger.cs

```diff
@@ -23,8 +27,21 @@ namespace Microsoft.Build.Logging.TerminalLogger;
 /// <remarks>
 /// Uses ANSI/VT100 control codes to erase and overwrite lines as the build is progressing.
 /// </remarks>
-internal sealed class TerminalLogger : INodeLogger
+internal sealed partial class TerminalLogger : INodeLogger
 {
+    private const string FilePathPattern = " -> ";
+
+#if NET7_0_OR_GREATER
+    [StringSyntax(StringSyntaxAttribute.Regex)]
+    private const string ImmediateMessagePattern = @"\[CredentialProvider\]|--interactive";
+    private const RegexOptions Options = RegexOptions.IgnoreCase | RegexOptions.CultureInvariant | RegexOptions.ExplicitCapture;
+
+    [GeneratedRegex(ImmediateMessagePattern, Options)]
+    private static partial Regex ImmediateMessageRegex();
+#else
+    private readonly string[] _immediateMessageKeywords = { @"\[CredentialProvider\]", "--interactive" };
```

Comment:
The backslashes are not interpreted as escape markers here, the strings will have literally `\[` in it. Also nit: `static`.

Reply:
This was correct before: the `\` in this case is a *regex* special character escaping `[`. Without the `\[\]` this collapses to "any character from the character class `[CredntialPov]`", instead of "the literal string `[CredentialProvider]`".

(it would have been valid to do `"\\[CredentialProvider\\]"` instead of `@"\[CredentialProvider\]"`)

Reply:
Oh wait I'm sorry, @ladipro was right because you switched to `IndexOf`. I'm surprised by the perf results there though, I see Regex being faster all the time (see https://github.com/dotnet/msbuild/pull/9411#discussion_r1392968936)

Reply:
I'm trying to understand the different perf results we got. It is quite intriguing.